src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -83,6 +83,50 @@ bool FCKey::operator==(const FCKey &rhs) const {
     return retVal;
 }
 
+class FCShapeInfer : public ShapeInferEmptyPads {
+public:
+    FCShapeInfer(size_t outPut_rank) : out_rank(outPut_rank) {}
+    std::vector<VectorDims> infer(
+        const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes,
+        const std::unordered_map<size_t, MemoryPtr>& data_dependency) override {
+        const VectorDims& activationShape = input_shapes[0].get();
+        const VectorDims& weightShape = input_shapes[1].get();
+        size_t activationRank = activationShape.size();
+        size_t channelRank = weightShape.size() - 1;
+        VectorDims outputShape;
+        // set batch dims
+        for (size_t i = 0; i < activationRank - channelRank; ++i) {
+            outputShape.push_back(activationShape[i]);
+        }
+        // append Co
+        outputShape.push_back(weightShape[0]);
+
+        while (outputShape.size() < out_rank) {
+            outputShape.insert(outputShape.begin(), 1);
+        }
```

Comment:
```suggestion
        VectorDims outputShape(out_rank, 1);
         // set Co
        outputShape.back() = weightShape[0];
        // set batch dims
        size_t batchDims = activationRank - channelRank
        size_t stratIdx = out_rank - (activationDims + 1);
        for (size_t i = 0; i < activationDims ; ++i) {
            outputShape[stratIdx ] = activationShape[i];
        }
```
Or something like that. The idea is to allocate enough memory at once to avoid possible internal buffer reallocations. Also, operation insert at the beginning over a std::vector means copy of n elements.

Reply:
done, thank you Maksim!